### PR TITLE
feat: automate inserting licenses on all python files

### LINKED ISSUE #99: Copyright notices are missing for some Python scripts
We have good coverage on copyright notices at the top of each files in https://github.com/couler-proj/couler/tree/d34a690/couler/core but many are still missing, for example:

* https://github.com/couler-proj/couler/tree/d34a690/couler/steps
* https://github.com/couler-proj/couler/tree/d34a690/couler/tests

### PR DESCRIPTION
**Pull request by xinbinhuang**
### What changes were proposed in this pull request?

Automate inserting licenses in all python files

Follow up & close #99

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Not required

### CODE REVIEW

**Review by terrytangyuan — COMMENTED** [review]
Thank you! Could you fix the build?

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+-r requirements.txt
```
Comment (xinbinhuang) [inline requirements-dev.txt:1]: This allows `pip install -r requirements-dev.txt` also installs `requirements.txt`

**Comment by xinbinhuang** [comment]
> Thank you! Could you fix the build?

Fixed!

**Review by terrytangyuan — APPROVED** [review]
Thanks!

**Comment by xinbinhuang** [comment]
> Thanks!

Thanks for reviewing!

**Pull request merged**